### PR TITLE
#15959 Extension attributes of quote on checkout page

### PR DESCRIPTION
**Pull request by skylineop**
Current pull request relates to this issue: https://github.com/magento/magento2/issues/15959 and should fix getting extension attributes, that assigned to a quote.

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=24771) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=24771) before we can accept your contribution.<br/><hr/>**skylineop** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

File: app/code/Magento/Checkout/Model/DefaultConfigProvider.php [hunk 1]
```diff
@@ -397,6 +397,9 @@ private function getQuoteData()
         if ($this->checkoutSession->getQuote()->getId()) {
             $quote = $this->quoteRepository->get($this->checkoutSession->getQuote()->getId());
             $quoteData = $quote->toArray();
+            if (is_object($quote->getExtensionAttributes())) {
```
Comment (sidolov) [inline app/code/Magento/Checkout/Model/DefaultConfigProvider.php]: I believe extension attributes can be an object or null, please replace is_object check comparison with null.
Also, please, cover changes with automated tests.

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/Model/DefaultConfigProvider.php (hunk 1 above)
Comment (skylineop) [reply app/code/Magento/Checkout/Model/DefaultConfigProvider.php]: Hi @sidolov, I have created additional commit with fixes, that relates to condition "is_object".
As for tests, I didn't find any tests concern operation getting quoteData. 
Thank you.

**Comment by sidolov** [comment]
@skylineop please, sign CLA

**Comment by skylineop** [comment]
> @skylineop please, sign CLA

Signed, thank you!

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-6127](https://jira.corp.magento.com/browse/ENGCOM-6127) has been created to process this Pull Request
:eight_spoked_asterisk: @VladimirZaets, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed.
PS: wouldn't it be worth of adding `if (is_object($quote->getExtensionAttributes()))` instead of `if (null !== $quote->getExtensionAttributes())`, just like it is with _$totals->getExtensionAttributes()_: 
```
if (is_object($totals->getExtensionAttributes())) {
            $totalsArray['extension_attributes'] = $totals->getExtensionAttributes()->__toArray();
}
```
There in CartInterface just like in TotalsInterface _getExtensionAttributes()_ method retrieves ether object or null:
```
/**
     * Retrieve existing extension attributes object or create a new one.
     *
     * @return \Magento\Quote\Api\Data\CartExtensionInterface|null
     */
    public function getExtensionAttributes();
```
At least it seems better for code chunks that are supposed to work similarly if at the same time they look as similar as possible.

**Comment by engcom-Golf** [comment]
@engcom-Bravo https://github.com/magento/magento2/pull/24771#discussion_r329218767

**Comment by engcom-Bravo** [comment]
Haven't noticed that @engcom-Golf)

**Pull request merged**